server.go

```diff
@@ -446,32 +1234,30 @@ func (s *server) handleQuery(querymsg interface{}, state *peerState) {
 			// however it is statistics for purely informational purposes
 			// and we don't really care if they are raced to get the new
 			// version.
-			p.StatsMtx.Lock()
 			info := &btcjson.GetPeerInfoResult{
-				ID:             p.id,
-				Addr:           p.addr,
-				Services:       fmt.Sprintf("%08d", p.services),
-				LastSend:       p.lastSend.Unix(),
-				LastRecv:       p.lastRecv.Unix(),
-				BytesSent:      p.bytesSent,
-				BytesRecv:      p.bytesReceived,
-				ConnTime:       p.timeConnected.Unix(),
-				TimeOffset:     p.timeOffset,
-				Version:        p.protocolVersion,
-				SubVer:         p.userAgent,
-				Inbound:        p.inbound,
-				StartingHeight: p.startingHeight,
-				CurrentHeight:  p.lastBlock,
+				ID:             p.ID(),
```

Comment:
In fact, now that peers are concurrent safe, this could be changed to simply return the list of peers and move all of the creation of the `btcjson.GetPeerInfoResult` into the rpc server where it belongs.